spotdl/check_down_ffmpeg.py

```diff
@@ -21,70 +21,48 @@ def download_ffmpeg():
-            # Since Desktop linux and Android both return "Linux", we need to do an additional check.
-            # I don't know and I have no way of testing if this is accurate.
-            # These are just educated guesses.
-            if platform.uname().machine == "i386":
-                #Desktop Linux
-                to_use_url = linux_url_i686
-            elif platform.uname().machine == "x86":
-                #Android
-                raise Exception("Sorry, 32-bit android is not supported by FFmpeg.")
-            else:
-                raise Exception("Could not detect architecture." +
-                " Please submit an issue on GitHub containing your system information.")
-        elif platform.system == "Darwin":
-            to_use_url = mac_url_32
+    sources = {
+        # !I used platform.machine() to find these values.
+        # !The 32bit values & Darwin's are educated guesses, as I do not have any 32bit machines
+        # !Or any macOS devices.
+        "Windows": {
+            "AMD64":    "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/win32-x64",
+            "i686":     "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/win32-ia32",
+            "arm":      "PUT SOMETHING HERE!!!!"
+        },
+        "Linux": {
+            "x86_64":   "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/linux-x64",
+            "x86":      "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/linux-ia32",
+            "arm32":    "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/linux-arm",
+            "aarch64":  "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/linux-arm64"
+        },
+        "Darwin": {
+            "x86_64":   "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/darwin-x64",
+            "x86":      "https://lame.buanzo.org/ffmpeg-mac-2.2.2.zip",
+            "arm":      "https://github.com/eugeneware/ffmpeg-static/releases/download/b4.4/darwin-arm64"
+        }
+    }
 
-    # TODO Create extract function for each architecture
+    try:
+        link_to_use = sources[platform.system()][platform.machine()]
+    except:
+        # Should change this message once we know for sure the output of the previous command
+        # On any and all OS and Architectures listed.
+        raise Exception("Could not detect architecture or operating system. "
+                        "Please open an issue on the GitHub page of this project.")
+    current_dir = os.path.dirname(__file__)
+    ffmpeg_data = requests.get(link_to_use, allow_redirects=True).content
+    if link_to_use == sources["Darwin"]["x86"]:
+        print("")
+        # TODO make functionality for this case
+    elif platform.system() == "Windows":
+        open(f"{current_dir}\\ffmpeg\\bin\\ffmpeg.exe", "wb").write(ffmpeg_data)
+    else:
+        open(f"{current_dir}/ffmpeg/bin/ffmpeg", "wb").write(ffmpeg_data)
```

Comment:
```suggestion
  	output_file = Path(
  	f"{current_dir}/ffmpeg/bin/ffmpeg{'.exe' if platform.system().lower() == 'windows' else ''}")
  		
  	with open(output_file, "wb") as file:
  		file.write(ffmpeg_data)
  		
  	# `chmod +x` ffmpeg executable on linux and mac
    if os in ["darwin", "linux"]:
    	output_file.chmod(output_file.stat().st_mode | stat.S_IEXEC)
  		
```

you will have to `import stat` and `from pathlib import Path`

don't commit it as is, use this as suggestion